python/metatensor-operations/metatensor/operations/_dispatch.py

```diff
@@ -136,6 +149,19 @@ def eye_like(array, size: int):
         return np.eye(size, dtype=array.dtype)
     else:
         raise TypeError(UNKNOWN_ARRAY_TYPE)
+    
+
+def full_like(shape: List[int], fill_value: Union[int, float], like):
+    """
+    Fills an array with a value. This will be a numpy array or 
+    a torch tensor based on the `like` argument.
+    """
+    if isinstance(like, TorchTensor):
+        return torch.full(shape, fill_value, dtype=like.dtype, device=like.device)
+    elif isinstance(like, np.ndarray):
+        return np.full(shape, fill_value)
+    else:
+        raise TypeError(UNKNOWN_ARRAY_TYPE)
 
 
 def list_to_array(array, data: Union[List[int], List[List[int]]]):
```

Comment:
should we rename this one to `int_array_like`?

Reply:
But it can also be a float, no?

Reply:
But it can also be a float, no?

Reply:
data is defined a `int`, and I don't think we ever call it with float

Reply:
Ok